pkg/agent/baker.go

```diff
@@ -95,6 +95,8 @@ func (t *TemplateGenerator) GetNodeBootstrappingCmd(config *datamodel.NodeBootst
 
 // getLinuxNodeCSECommand returns Linux node custom script extension execution command
 func (t *TemplateGenerator) getLinuxNodeCSECommand(config *datamodel.NodeBootstrappingConfiguration) string {
+	// validate and fix input
+	validateAndSetLinuxNodeBootstrappingConfiguration(config)
```

Comment:
this fixes the dynamic config dir issue, otherwise the new unit tests break because I added the deepcopy to avoid mutation in baker_test.go